src/lsm/forest.zig

```diff
@@ -328,7 +326,16 @@ pub fn ForestType(comptime Storage: type, comptime groove_cfg: anytype) type {
                     const grooves: *align(@alignOf(Grooves)) Grooves =
                         @alignCast(@fieldParentPtr(Grooves, groove_field_name, groove));
                     const forest = @fieldParentPtr(Forest, "grooves", grooves);
-                    forest.progress.?.compact.pending -= 1;
+
+                    inline for (std.meta.fields(Grooves), 0..) |groove_field, i| {
+                        if (std.mem.eql(u8, groove_field.name, groove_field_name)) {
```

Comment:
:thinking: I think this generates a runtime comparison (which could be optimized away by LLVM, but I think the semantics here is runtime). 

I think we want rather something like:

```
const groove_index = comptime for ((std.meta.fields(Grooves), 0..)) |groove_field, i| {
    if (std.mem.eql(u8, groove_field.name, groove_field_name)) break i;
} else unreachable;
```